packages/device_info_plus/device_info_plus_platform_interface/lib/device_info_plus_platform_interface.dart

```diff
@@ -49,39 +38,8 @@ abstract class DeviceInfoPlatform extends PlatformInterface {
     _instance = instance;
   }
 
-  // Gets the Android device information.
   // ignore: public_member_api_docs
-  Future<AndroidDeviceInfo> androidInfo() {
-    throw UnimplementedError('androidInfo() has not been implemented.');
-  }
-
-  // Gets the iOS device information.
-  // ignore: public_member_api_docs
-  Future<IosDeviceInfo> iosInfo() {
-    throw UnimplementedError('iosInfo() has not been implemented.');
-  }
-
-  // Gets the Linux device information.
-  // ignore: public_member_api_docs
-  Future<LinuxDeviceInfo> linuxInfo() {
-    throw UnimplementedError('linuxInfo() has not been implemented.');
-  }
-
-  // Gets the web browser information.
-  // ignore: public_member_api_docs
-  Future<WebBrowserInfo> webBrowserInfo() {
-    throw UnimplementedError('webBrowserInfo() has not been implemented.');
-  }
-
-  // Gets the Macos device information.
-  // ignore: public_member_api_docs
-  Future<MacOsDeviceInfo> macosInfo() {
-    throw UnimplementedError('macosInfo() has not been implemented.');
-  }
-
-  // Gets the Windows device information
-  // ignore: public_member_api_docs
-  Future<WindowsDeviceInfo>? windowsInfo() {
-    throw UnimplementedError('windowsInfo() has not been implemented.');
```

Comment:
All platform specific methods are now replaced by a single `deviceInfo()` making the platform interface generic